miner/util.go

```diff
@@ -10,14 +10,30 @@ import (
 	"github.com/spacemeshos/go-spacemesh/sql/ballots"
 	"github.com/spacemeshos/go-spacemesh/sql/blocks"
 	"github.com/spacemeshos/go-spacemesh/sql/certificates"
+	"github.com/spacemeshos/go-spacemesh/sql/layers"
 )
 
-func ActiveSetFromEpochFirstBlock(db sql.Executor, epoch types.EpochID) ([]types.ATXID, error) {
+func ActiveSetFromEpochFirstCertifiedBlock(db sql.Executor, epoch types.EpochID) ([]types.ATXID, error) {
 	bid, err := certificates.FirstInEpoch(db, epoch)
 	if err != nil {
 		return nil, err
 	}
+	return activeSetFromBlock(db, bid)
+}
+
+func ActiveSetFromEpochFirstBlock(db sql.Executor, epoch types.EpochID) ([]types.ATXID, error) {
+	result, err := ActiveSetFromEpochFirstCertifiedBlock(db, epoch)
```

Comment:
this would be a different  block for synced node, i think it should be always first applied regardless of certificate